lib/doc/row.js

```diff
@@ -367,6 +367,6 @@ Row.prototype = {
     this.hidden = value.hidden;
     this.outlineLevel = value.outlineLevel || 0;
 
-    this.style = value.style || {};
+    this.style = value.style && JSON.parse(JSON.stringify(value.style)) || {};
```

Comment:
fix is here ;)

Reply:
I'm not a big fan of this, since there's a couple of subtle ways this can fail or lose things depending on the shape of the object. do we know what values each key of `value.style` can hold? is it just a flat object of `"some string": "another string"` or can it also hold numbers, arrays, functions or other objects?

Reply:
it is only object that keep [configuration of styles](https://github.com/exceljs/exceljs#styles)
there aren't any functions or getters/setters

Reply:
gotcha. Can you help me understand the performance impact here?
`JSON.parse(JSON.stringify())` is quite slow and it would suck if this actually runs for all (or most) rows. Also, it increases the memory overhead a bit. How many rows will have this property?

Reply:
personally I think it's much better to write style cloning function in utils ;)

Reply:
Using JSON is indeed pretty slow to copy things. You could use e.g., https://github.com/davidmarkclements/rfdc for now (I would normally just recommend to use object spread but that's not available in versions below 8 and slow before v.10 or v.11 :smile:).

@alubbe the current API does not seem to have performance in mind. If the user changes any input that should be registered in a clean way so all internals can work side effect free. This does not seem to be possible without the penalty here. Using getters and setters might reduce the overhead by only copying values that have to be changed but as far as I see it that would probably be pretty brittle with the current code.

Reply:
I agree - let's merge it and then actually benchmark why this library is slow and fix those, as opposed to pre-optimizing a bugfix